src/nodedumper.cc

```diff
@@ -26,11 +26,10 @@ bool NodeDumper::isCached(const AbstractNode &node) const
 void NodeDumper::handleIndent(const State &state)
 {
 	if (state.isPrefix()) {
-		this->currindent += "\t";
+		this->currindent++;
 	}
 	else if (state.isPostfix()) {
-		this->currindent.erase((this->currindent.length() >= 1) ? 
-													 this->currindent.length() - 1 : 0);
+		this->currindent = this->currindent > 0 ? this->currindent - 1 : 0;
```

Comment:
Now that I read this code, I don't think `currindent` can ever become negative, so I think `this->currindent--;` should be sufficient